### PR TITLE
Adding an option to plot norm values of the grads in plot_evoked_topo

### LINKED ISSUE #2884: Adding an option to plot norm values of the grads in plot_evoked_topo?
Hi,
It would be great if an option to plot norm values of combined gradiometer pairs (sqrt of (grad12 + grad22)) could be added in plot_evoked_topo. Brain storm does that automatically in its gui (http://neuroimage.usc.edu/brainstorm/Tutorials/TutMindNeuromag) and various groups seem to publish such timecourses of the combined grads.
Much thanks for your help!
Maria

### PR DESCRIPTION
**Pull request by cjayb**
Well, maybe not so hard after all... Here is a rough proof-of-concept. Needs to be cleaned up with decisions on how to layout when `merge_grads` is requested (new layout for combined grads based on mag?) + implementation

Could use some guidance here, treading unknown areas of the codebase...

eventually:
closes #2884

### CODE REVIEW

File: mne/viz/topo.py [hunk 1]
```diff
@@ -350,9 +351,20 @@ def _plot_evoked_topo(evoked, layout=None, layout_scale=0.945, color=None,
...
-            e.data[pick] = e.data[pick] * scalings[t]
+        for pick, ch_type in zip(picks, types_used):
+            e.data[pick] = e.data[pick] * scalings[ch_type]
+
+            if ch_type == 'grad' and merge_grads:
+                from ..channels.layout import _merge_grad_data
+
+                # HACK replace mag data with grad_norm!
+                # seems to be doing something meaningful?
+                e.data[picks[0]] = _merge_grad_data(e.data[pick])
```
Comment (dengemann) [inline mne/viz/topo.py]: I think what you might want to to make it cleaner, is to copy evoked and use .add_channels to add the normed data to a new evoked object. Then of course the scaling would not apply any longer.

Comment (jaeilepp) [inline mne/viz/topo.py]: And you probably want to call ``_pair_grad_sensors`` before ``_merge_grad_data``.

**Comment by agramfort** [comment]
@jaeilepp you can take over if you have time.

**Comment by cjayb** [comment]
> @jaeilepp you can take over if you have time.

+1, probably a case where it's a lot faster to do it yourself than wait that I "get it"...

It's still a little unclear (to me) what the desired API is: given a VV dataset, what is the plot to look like? 'mag' and 'norm_grad' side-by-side? Just one or the other (hack a 'Vectorview-mag' layout for the norms)?

**Comment by Eric89GXL** [comment]
> Just one or the other (hack a 'Vectorview-mag' layout for the norms)?

This one

**Comment by cjayb** [comment]
this commit is FYI, just in case any of it useful. bewarned: hacky as hell! HTH for discussion, if nothing else...

The code I use to test is from `plot_topography.py`, here for quick reference:
```
%matplotlib qt
import matplotlib.pyplot as plt
from mne import read_evokeds
from mne.viz import plot_evoked_topo
from mne.datasets import sample
data_path = sample.data_path()
fname = data_path + '/MEG/sample/sample_audvis-ave.fif'
condition = 'Left Auditory'
evoked = read_evokeds(fname, condition=condition, baseline=(None, 0))
title = 'MNE sample data (condition : %s)' % evoked.comment
plot_evoked_topo(evoked, title=title)
```

**Pull request closed without merging**